### DISCUSSION TITLE
Add label names filter to pull_request_target on labels

### DISCUSSION THREAD

**Discussion by yahavi** [community/community#14407] [Actions] [answered]
Please inspect the following workflow:
```yaml
name: "Test"

on:
  pull_request_target:
    types: [labeled]
    labels: # <- Please add this feature 🙏🏼 
      - 'safe to test'

jobs:
  tests:
    if: contains(github.event.pull_request.labels.*.name, 'safe to test')
    steps: ...
```

I'd like my workflow to run after creating the 'safe to test' label. This workflow runs after adding EVERY label.
If I add 'safe to test' then it runs the tests as expected.
If I add a different label, the workflow will start to run but the job will be skipped. This will cause the "tests" workflow to be marked as skipped, despite the fact that I already ran them before.

To summarize - I'd like that a workflow will be running only after adding a SPECIFIC label, and not after EVERY label.
Thanks! 🙏🏼

**Comment by ethomson** [comment, marked as answer]
Thanks for the feedback @yahavi. At the moment, as you note, there's no way to handle this at the workflow trigger level, but I'll put this on the backlog.

**Reply by yahavi** [reply]
@ethomson thanks for putting this in the backlog. Looking forward to it!

**Reply by Frans-L** [reply]
Hey! Any update on this? :)

**Comment by soemo** [comment]
I would also like to have this feature. Is there currently a workaround?

**Reply by polsala** [reply]
"yes", using an auxiliary "check".

For example something like this:

```yml
on:
  push:
  pull_request:
    types: [ labeled ]
  
jobs:
  tests:
    if: ${{ (github.event_name == 'pull_request' && github.event.label.name == 'xxxxxxxxxxxxxx') }}
    runs-on: [self-hosted, xxxxxxx]
    steps:
      ...multiple steps
      - name: Set aux check
        if: steps.xxxxx.outcome == 'success'
        uses: ouzi-dev/commit-status-updater@v2
        with:
          name: "Aux Check name"
          status: "${{ steps.xxxxx.outcome }}"
          url: "https://github.com/${{ github.repository }}/actions/runs/${{ github.run_id }}"
```
With the auxiliary check you will get it to only set when you do not skip. You also have to keep an eye on the concurrency so that it is not interrupted, for example using `github.event.label.name`.
You can create check manually with gh too instance of precooked action.

**Comment by axel-h** [comment]
Any update here, when this feature would be available? Or alternatively, having `if:` on the top level is also an improvement.

**Comment by untereiner** [comment]
Hello, I picked you up @joshuawalker in another thread sorry. What's the status of this improvement in the backlog ? Can we hope using it someday ?

**Comment by rostislav-simonik-plc** [comment]
This could also reduce costs, as we currently need to check for the presence of a PR label at runtime, and GitHub charges for any execution taking less than a minute as a full minute.

**Comment by gtjoseph** [comment]
We've been struggling with this issue for almost two years now.  The concurrency issue isn't trivial.

Let's say you have two workflows triggered by "labeled"...
A: whose job runs on label "A" and has concurrency group A-${{ github.event.number }}.
B: whose job runs on label "B" and has concurrency group B-${{ github.event.number }}.

If you add label "A" to the PR, both workflows concurrency groups are evaluated before A's job runs and B's job decides it skips.
If you add label "B" to the PR before "A" finishes, again both workflows concurrency groups are evaluated and B's job runs BUT the previously running "A" workflow is cancelled!

NOTE: You HAVE to use concurrency at the workflow level because using a concurrency block on a job doesn't properly evaluate github.event.number.
